Some notes regarding the API.

BigETI opened the pull request.

BigETI committed:
Some notes regarding the API.

Southclaws commented on NOTES.md:
```diff
@@ -0,0 +1,95 @@
+- Multiple requests in a short period of time can be rate-limited by the backend RESTful API to prevent heavy workload.
+- Servers can't subscribe other servers.
+- The backend RESTful API can obtain IP addresses from the San Andreas Multiplayer API, and obtain server data accordingly.
+
```
This is the important stuff I usually forget to write, thanks!

Southclaws commented on NOTES.md:
```diff
@@ -0,0 +1,95 @@
+- Lifetime: Number of seconds to life as integer, -1 for default lifetime
+- Extended: This attribute is an object that cotains multiple attributes like for example website URL, contact email and etc.
+
+The message is contained in a HTTP GET request "/subscribe"
```
I was planning on just using the `/server/{address}` endpoint with a POST method for this (just to simplify the amount of routes - both of these actions are getting and setting the same information so it makes sense to use the same endpoint). also "subscribe" usually means to request information periodically rather than to send or "announce" it.

Southclaws commented on NOTES.md:
```diff
@@ -0,0 +1,95 @@
+
+The communication between the backend RESTful API and the server uses JSON (JavaScript Object Notation) that consists of these attributes:
+
+- Status: The current status code as an integer, see HTTP status codes
```
HTTP response includes this header implicitly so there would be no need to duplicate it into the JSON payload.

Southclaws commented on NOTES.md:
```diff
@@ -0,0 +1,95 @@
+- Hostname: Hostname of the server
+- Gamemode: Gamemode of the server
+- Language: Language of the server
+- MaxPlayers: Maximum amount of players that the server supports
```
I did forget about player counts, thanks for reminding me!

Southclaws commented on NOTES.md:
```diff
@@ -0,0 +1,95 @@
+- Gamemode: Gamemode of the server
+- Language: Language of the server
+- MaxPlayers: Maximum amount of players that the server supports
+- Extended: This attribute is an object that cotains multiple attributes like for example website URL, contact email and etc.
```
I think extended information only needs to be retrieved on a call to the "server details" endpoint, the main list should be as minimal as possible if it's going to respond with every server in a single object (or x amount of servers if it's paged).

Southclaws reviewed (changes requested):
Thanks for the contribution, I'm glad there's interest in this project! I've put some comments so we can discuss the schema a little more before finalising the first version.

BigETI committed:
Changed to request

Replaced endpoint "/subscribe" with "/servers/<domain name or IPv4 address and port>", renamed subscription to announcement, removed status code attribute from request, removed extended attribute from response by GET request "/servers".

BigETI commented:
I've changed some things on my latest commit accordingly.

Southclaws commented:
Thanks! I'll start the first version and chat about it on Discord at some point :+1:
